boring/src/ssl/mod.rs

```diff
@@ -692,17 +728,38 @@ pub struct SslContextBuilder {
     is_rpk: bool,
 }
 
+unsafe fn new_ssl_ctx_with_default_curves(
+    method: SslMethod,
+) -> Result<*mut ffi::SSL_CTX, ErrorStack> {
+    init();
+    let ctx = cvt_p(ffi::SSL_CTX_new(method.as_ptr()))?;
+    ssl_ctx_set_curves(ctx, SslCurve::DEFAULT_CURVES)?;
```

Comment:
instead of using ffi, we can use safe API on builder: https://docs.rs/boring/latest/boring/ssl/struct.SslContextBuilder.html#method.set_curves

Reply:
So the reason I'm using the unsafe API here is that I want to call `SSL_CTX_set1_curves()` wherever we call `SSL_CTX_new()`. If you look at the constructors for `SslContextBuilder`, they all use the unsafe API. I coiuld make `new_ssl_ctx_with_default_curves()` safe, but that would require more wrangling because `SslContextBuilder` also has the `is_rpk` flag.

Also, note that `ssl_ctx_set_curves()` is now also used by `SslContextBuilder::set_curves()`.

Reply:
E.g. instead of:
```rust 
    pub fn new_rpk() -> Result<SslContextBuilder, ErrorStack> {
        unsafe {
            init();
            let ctx = cvt_p(ffi::SSL_CTX_new(SslMethod::tls_with_buffer().as_ptr()))?;

            Ok(SslContextBuilder::from_ptr(ctx, true))
        }
    }
```
    have
```rust
    pub fn new_rpk() -> Result<SslContextBuilder, ErrorStack> {
        let mut ctx = unsafe {
            init();
            let ctx = cvt_p(ffi::SSL_CTX_new(SslMethod::tls_with_buffer().as_ptr()))?;

            SslContextBuilder::from_ptr(ctx, true)
        };

       ctx.set_curves(SslCurve::DEFAULT_CURVES)?;

        Ok(ctx)
    }
```